lib/factorial.rb

```diff
@@ -1,6 +1,13 @@
 # Computes factorial of the input number and returns it
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: O(n) where n is the value of the input number
+# Space complexity: O(1)
```

Comment:
Actually the space complexity is O(n) because every time you make a method call, and you're making ~n of them, Ruby puts the current method in memory on something called the stack.